### PR TITLE
Development guide and unified functions

### LINKED ISSUE #1120: Create readme for unattended installer
We need to create a readme for this folder. We will put there all links that helped on this development asides from how this installer works.

### PR DESCRIPTION
**Pull request by alberpilot**
Closes #1120
This PR aims to unify some functions, adds some clarifications, and a development guide.

### CODE REVIEW

File: tests/unattended/unit/suites/test-elasticsearch.sh [hunk 1]
```diff
@@ -37,6 +32,7 @@ test-copyCertificatesElasticsearch() {
 }
 
 test-copyCertificatesElasticsearch-assert() {
+    mkdir /etc/elasticsearch/certs/
```
Comment (DFolchA) [inline tests/unattended/unit/suites/test-elasticsearch.sh:35]: Use `mkdir -p` to make sure that the path exists.

File: tests/unattended/unit/suites/test-elasticsearch.sh [hunk 2]
```diff
@@ -133,23 +129,22 @@ test-configureElasticsearch-dist-one-elastic-node() {
 
 
 test-configureElasticsearch-dist-one-elastic-node-assert() {
-    getConfig elasticsearch/elasticsearch_unattended_distributed.yml /etc/elasticsearch/elasticsearch.yml
     getConfig elasticsearch/roles/roles.yml /usr/share/elasticsearch/plugins/opendistro_security/securityconfig/roles.yml
     getConfig elasticsearch/roles/roles_mapping.yml /usr/share/elasticsearch/plugins/opendistro_security/securityconfig/roles_mapping.yml
     getConfig elasticsearch/roles/internal_users.yml /usr/share/elasticsearch/plugins/opendistro_security/securityconfig/internal_users.yml
 
     rm /etc/elasticsearch/esnode-key.pem /etc/elasticsearch/esnode.pem /etc/elasticsearch/kirk-key.pem /etc/elasticsearch/kirk.pem /etc/elasticsearch/root-ca.pem -f
```
Comment (DFolchA) [inline tests/unattended/unit/suites/test-elasticsearch.sh:136]: You can use rm /etc/elasticsearch/{name1,name2,...}`

File: tests/unattended/unit/suites/test-elasticsearch.sh [hunk 3]
```diff
@@ -167,55 +162,56 @@ test-configureElasticsearch-dist-two-elastic-nodes() {
...
     copyCertificatesElasticsearch
 
     sed -i 's/-Xms1g/-Xms1g/' /etc/elasticsearch/jvm.options
     sed -i 's/-Xmx1g/-Xmx1g/' /etc/elasticsearch/jvm.options
 
+    getConfig elasticsearch/elasticsearch_all_in_one.yml /etc/elasticsearch/elasticsearch.yml
+
     applyLog4j2Mitigation
+
+    rm /etc/elasticsearch/certs/client-certificates.readme /etc/elasticsearch/certs/elasticsearch_elasticsearch_config_snippet.yml -f
```
Comment (DFolchA) [inline tests/unattended/unit/suites/test-elasticsearch.sh:213]: You can use rm /etc/elasticsearch/{name1,name2,...}`, and please place the `-f` at the start

File: unattended_installer/install_functions/opendistro/common.sh [hunk 4]
```diff
@@ -355,49 +355,26 @@ function rollBack() {
...
-    fi
-
-    if [ -f "/lib/firewalld/services/kibana.xml" ]; then
-        eval "rm -f /lib/firewalld/services/kibana.xml ${debug}"
-    fi
-
-    if [ -f "/lib/firewalld/services/elasticsearch.xml" ]; then
-        eval "rm -f /lib/firewalld/services/elasticsearch.xml ${debug}"
-    fi
+    for directory_to_remove in "${directories_to_remove[@]}"; do
```
Comment (DFolchA) [inline unattended_installer/install_functions/opendistro/common.sh:362]: Is this loop necessary can't you just run:
```
rm -rf "${directories_to_remove[*]}"
```

File: unattended_installer/install_functions/opendistro/common.sh [hunk 5]
```diff
@@ -355,49 +355,26 @@ function rollBack() {
...
+    files_to_remove=( "/securityadmin_demo.sh"
+                      "/etc/systemd/system/multi-user.target.wants/wazuh-manager.service"
+                      "/etc/systemd/system/multi-user.target.wants/filebeat.service"
+                      "/etc/systemd/system/multi-user.target.wants/elasticsearch.service"
+                      "/etc/systemd/system/multi-user.target.wants/kibana.service"
+                      "/etc/systemd/system/kibana.service"
+                      "/lib/firewalld/services/kibana.xml"
+                      "/lib/firewalld/services/elasticsearch.xml" )
+
+    for file_to_remove in "${files_to_remove[@]}"; do
```
Comment (DFolchA) [inline unattended_installer/install_functions/opendistro/common.sh:375]: Same

**Review by DFolchA — CHANGES_REQUESTED** [review]

File: tests/unattended/unit/suites/test-elasticsearch.sh (hunk 2 above)
Comment (verdx) [reply tests/unattended/unit/suites/test-elasticsearch.sh]: Done https://github.com/wazuh/wazuh-packages/pull/1188/commits/107118c34312b821527a77cfc9dfbc175bcb14e4

File: tests/unattended/unit/suites/test-elasticsearch.sh (hunk 3 above)
Comment (verdx) [reply tests/unattended/unit/suites/test-elasticsearch.sh:212]: Done https://github.com/wazuh/wazuh-packages/pull/1188/commits/107118c34312b821527a77cfc9dfbc175bcb14e4

File: tests/unattended/unit/suites/test-elasticsearch.sh (hunk 1 above)
Comment (verdx) [reply tests/unattended/unit/suites/test-elasticsearch.sh]: Done https://github.com/wazuh/wazuh-packages/pull/1188/commits/0d4d12564d2435c1178c68241eed31a20cc52250

File: unattended_installer/Development-guide.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+This program is free software; you can redistribute it
+and/or modify it under the terms of the GNU General Public
+License (version 2) as published by the FSF - Free Software
+Foundation.
+
+## Useful links and acknowledgment
+
+- [Bash meets solid](https://codewizardly.com/bash-meets-solid/)
+- [Shellcheck](https://github.com/koalaman/shellcheck#gallery-of-bad-code)
```
Comment (verdx) [inline unattended_installer/Development-guide.md:46]: Add bach.sh acknowledgement

**Review by verdx — CHANGES_REQUESTED** [review]

**Review by DFolchA — APPROVED** [review]
LGTM

**Pull request merged**